cli/internal/client/client.go

```diff
@@ -143,11 +145,23 @@ func (c *ApiClient) PutArtifact(hash string, duration int, rawBody interface{})
 	if encoded != "" {
 		encoded = "?" + encoded
 	}
-	req, err := retryablehttp.NewRequest(http.MethodPut, c.makeUrl("/v8/artifacts/"+hash+encoded), rawBody)
+	// Read the entire artifactReader into memory so we can easily compute the Content-MD5.
+	// Note: retryablehttp.NewRequest reads the artifactReader into memory so there's no
+	// additional overhead by doing the ioutil.ReadAll here instead.
+	artifactBody, err := ioutil.ReadAll(artifactReader)
+	if err != nil {
+		return fmt.Errorf("failed to store files in HTTP cache: %w", err)
+	}
+	md5Sum := md5.Sum(artifactBody)
+	contentMd5 := base64.StdEncoding.EncodeToString(md5Sum[:])
+
+	req, err := retryablehttp.NewRequest(http.MethodPut, c.makeUrl("/v8/artifacts/"+hash+encoded), artifactBody)
```

Comment:
@gsoltis , the backend probably needs to merge first before we remove the content-length is defined assumption. 

I think this is what led to some errors today when this tried to merge. Going to think of a better way to handle the chunked scheme. in the meantime I added the content-md5 to the header.

Reply:
cc @jaredpalmer